### ISSUE TITLE
Connection Timeout when running `make`

### ISSUE THREAD

**Issue by exhuma** [neovim/neovim#1830]
I just cloned from the master branch (5c6348e) and ran the following:

    make CMAKE_EXTRA_FLAGS="-DCMAKE_INSTALL_PREFIX:PATH=/opt/neovim

It ended with the following error:

    [100%] Generating usr/lib/luarocks/rocks/nvim-client
    Cloning into 'lua-client'...
    fatal: unable to connect to github.com:
    github.com[0: 192.30.252.129]: errno=Connection timed out

    Error: Failed cloning git repository.
    make[3]: *** [usr/lib/luarocks/rocks/nvim-client] Error 1
    make[3]: Leaving directory `/home/users/michel/work/public/neovim/.deps'
    make[2]: *** [CMakeFiles/nvim-client.dir/all] Error 2
    make[2]: Leaving directory `/home/users/michel/work/public/neovim/.deps'
    make[1]: *** [all] Error 2
    make[1]: Leaving directory `/home/users/michel/work/public/neovim/.deps'
    make: *** [deps] Error 2

I *assume* this is because this is run from behind a corp. firewall which only allows HTTP(S) connections via a proxy. Both `http_proxy` and `https_proxy` are set and valid (`curl https://www.google.com` works). Is it possible that there is a line somewhere in the build process which uses the `git://` protocol? If yes, would it not be better to use `https` instead as this is supported more widely?

I grepped for both `github.com` and `lua-client` to find the line causing this error, but could not find it.

**Comment by justinmk** [comment]
I don't think I've seen a timeout before, but similar issues here: https://github.com/neovim/neovim/issues/1703 https://github.com/neovim/neovim/issues/1469

tl;dl: make sure cmake is compiled with SSL support.

**Comment by exhuma** [comment]
I am not sure if it's an SSL error. The error message above does not include the protocol. It could be SSL, but it could be something else as well.

**Comment by Pyrohh** [comment]
Try running `ld /usr/bin/cmake | grep ssl' ,or wherever your cmake binary is installed. The output should be something like this, at least on linux:
```
libssl.so.1.0.0 => /usr/lib/libssl.so.1.0.0 (0x00007f5825c8b000)
```
If that's not the case then something might have gone wrong during the linking stage.
Have you  already tried [this comment's instructions](https://github.com/neovim/neovim/issues/1469#issuecomment-63058312)?

**Comment by dzeban** [comment]
**TL;DR**

    git config --global http.proxy http://<user>:<pass>@<proxy>:<port>
    git config --global url.https://github.com/.insteadOf git://github.com/

I had the same problem and found solution. There is [a rockspec](https://raw.githubusercontent.com/neovim/lua-client/a6c232da23dafe085c5606a3d0a666a77dd7d8bd/nvim-client-0.0.1-7.rockspec)  that contains URL for nvim-client - `git://github.com/neovim/lua-client`. If you're behind some sort of firewall or proxy you're probably forbidden to use git protocol. If you try to issue

    git clone git://github.com/neovim/lua-client

you will get that timeout error message. If you try same URL but use https protocol like this

    git clone https://github.com/neovim/neovim.git

and succeeds then you need to configure 2 things:

1. Configure proxy for git. 
2. Configure rewriting `git://` urls to `https://` urls

For simple proxies you can just configure it with `git config http.proxy`. If you have some exotic setup `gitProxy` scripts could help you - see examples [here](https://wiki.yoctoproject.org/wiki/Working_Behind_a_Network_Proxy).

Now that you can connect via proxy you need to pass through proxy by using `https` instead of `git` protocol. There is an config option `insteadOf` for this:

    git config --global url.https://github.com/.insteadOf git://github.com/

This will substitute `git://` urls for `https://` urls for all github repos.

After that I had successfully built neovim (`make distclean` and `make`).

Here are these options for .gitconfig:
````
[http]
	proxy =  http://<user>:<pass>@<proxy>:<port>
[url "https://github.com/"]
	insteadOf = git://github.com/
````

**Comment by exhuma** [comment]
Thanks @dzeban. This solved the issue. I was unaware of `url.insteadOf`. I'll keep that in mind!

**Comment by justinmk** [comment]
@dzeban Thanks for debugging that. Why don't we change the rockspec to https?

**Comment by BatmanAoD** [comment]
@dzeban Thank you so much.

**Comment by dzeban** [comment]
@justinmk Though problem is not with rockspec but with git configuration, IMHO it makes more sense to use https for fetching third-party stuff.

**Comment by exhuma** [comment]
@justinmk I suppose that's a good common ground. I'm sure the `https` protocol is much more accessible in general to anyone. Primarily due to corporate firewalls. As it's supported by github anyhow, I don't think that change would hurt anyone.

**Comment by lambdalisue** [comment]
:+1: I faced same problem and it was difficult to figure out the reason thus I prefer `https`

**Comment by fwalch** [comment]
Created https://github.com/neovim/lua-client/issues/10.

**Comment by lambdalisue** [comment]
:+1:

**Comment by lambdalisue** [comment]
https://github.com/neovim/neovim/blob/master/third-party/cmake/BuildLuarocks.cmake#L97

This should be updated as well.

**Comment by SergioGeeK7** [comment]
thanks @dzeban
